net-p2p/bitcoin-core/bitcoin-core-25.0.ebuild

```diff
@@ -0,0 +1,251 @@
+		$(use_with dbus qtdbus)
+		$(use_with system-leveldb)
+		$(use_with system-libsecp256k1)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	use external-signer && DOCS+=( doc/external-signer.md )
+	use berkdb || use sqlite && DOCS+=( doc/managing-wallets.md )
+	use libs && DOCS+=( doc/shared-libraries.md )
+	use systemtap && DOCS+=( doc/tracing.md )
+	use zeromq && DOCS+=( doc/zmq.md )
+
+	default
+
+	! use libs || find "${ED}" -name '*.la' -delete || die
+	! use test || rm -f -- "${ED}"/usr/bin/test_bitcoin{,-qt} || die
+
+	newbashcomp contrib/completions/bash/bitcoin-tx.bash-completion bitcoin-tx
+	use bitcoin-cli && newbashcomp contrib/completions/bash/bitcoin-cli.bash-completion bitcoin-cli
+	use daemon || use qt5 && newbashcomp contrib/completions/bash/bitcoind.bash-completion bitcoind
+
+	if use daemon ; then
+		insinto /etc/bitcoin
+		sed -ne 's/ To use, copy this file$//p;Tp;:0;n;/save the file\.$/!b0;d;:p;p' \
+			share/examples/bitcoin.conf >"${ED}/etc/bitcoin/bitcoin.conf" || die
+		fowners bitcoin:bitcoin /etc/bitcoin/bitcoin.conf
+		fperms 600 /etc/bitcoin/bitcoin.conf
+
+		newconfd contrib/init/bitcoind.openrcconf bitcoind
+		newinitd "${FILESDIR}/bitcoind.openrc" bitcoind
+		systemd_newunit contrib/init/bitcoind.service bitcoind.service
+
+		keepdir /var/lib/bitcoind
+		fperms 700 /var/lib/bitcoind
+		fowners bitcoin:bitcoin /var/lib/bitcoind
+		dosym -r /etc/bitcoin/bitcoin.conf /var/lib/bitcoind/bitcoin.conf
+		dosym -r /var/lib/bitcoind /var/lib/bitcoin/.bitcoin
+
+		doman "${FILESDIR}/bitcoin.conf.5"
+
+		insinto /etc/logrotate.d
+		newins "${FILESDIR}/bitcoind.logrotate-r1" bitcoind
+	fi
+
+	if use qt5 ; then
+		insinto /usr/share/icons/hicolor/scalable/apps
+		newins src/qt/res/src/bitcoin.svg bitcoin128.svg
+
+		domenu "${FILESDIR}/org.bitcoin.bitcoin-qt.desktop"
+
+		if use kde ; then
+			insinto /usr/share/kservices5
+			doins "${FILESDIR}/bitcoin-qt.protocol"
+			dosym ../../kservices5/bitcoin-qt.protocol /usr/share/kde4/services/bitcoin-qt.protocol
+		fi
+	fi
+
+	if use examples ; then
+		docinto examples
+		dodoc -r contrib/{linearize,qos}
+		use zeromq && dodoc -r contrib/zmq
+	fi
+}
+
+pkg_preinst() {
+	if use daemon ; then
+		if [[ ! -d "${EROOT}/var/lib/bitcoin/.bitcoin" ]] ; then
+			rm -r -- "${ED}/var/lib/bitcoin" || die
+		elif [[ ! -e "${EROOT}/var/lib/bitcoind" ]] ; then
+			elog "Moving your ${EPREFIX}/var/lib/bitcoin/.bitcoin to ${EPREFIX}/var/lib/bitcoind."
+			mv --no-clobber --no-target-directory -- "${EROOT}/var/lib/bitcoin"{/.bitcoin,d} || die
+			elog 'A transitional symlink will be installed for your convenience.'
+		fi
+	fi
+
+	if use kde && [[ ! -d "${EROOT}/usr/share/kde4" ]] ; then
+		rm -r -- "${ED}/usr/share/kde4" || die
+	fi
+}
+
+pkg_postinst() {
+	# we don't use xdg.eclass because it adds unconditional IDEPENDs
+	if use qt5 ; then
+		xdg_desktop_database_update
+		xdg_icon_cache_update
+	fi
+
+	elog "To have ${PN} automatically use Tor when it's running, be sure your"
```

Comment:
Maybe only emit this message if `use daemon` (and maybe only on new installs)?

Reply:
This isn't a new log message. I just copied it over from the existing ebuild.

Does it not apply to Bitcoin-Qt as well? (I don't actually know, as I've never used Tor or Bitcoin-Qt myself.)

Reply:
> Maybe only emit this message block if `use daemon` and maybe only on new installs, for example by checking `REPLACING_VERSIONS`

I made this change.